tests/compile-fail/lifetimes.rs

```diff
@@ -0,0 +1,70 @@
+#![feature(plugin)]
+#![plugin(clippy)]
+
+#![deny(needless_lifetimes)]
+
+fn distinct_lifetimes<'a, 'b>(_x: &'a u8, _y: &'b u8, _z: u8) { } //~ERROR
```

Comment:
Totally agreed, will use that here, and also in my other recently written tests.